test/common/tracing/BUILD

```diff
@@ -8,47 +8,11 @@ licenses(["notice"])  # Apache 2
 
 envoy_package()
 
-envoy_cc_test(
-    name = "http_tracer_impl_legacy_test",
```

Comment:
What is the difference between `http_tracer_impl_test` and `http_tracer_impl_legacy_test`?

Reply:
runtime feature is enabled in  `http_tracer_impl_test `while is not enabled in `http_tracer_impl_legacy_test`. Since runtime guard is now removed, I think we should also remove the legacy test.